### ISSUE TITLE
[JOSS] Installation

### ISSUE THREAD

**Issue by RMeli** [ur-whitelab/hoomd-tf#184]
Hello! I'll open some issues as part of the [JOSS review](https://github.com/openjournals/joss-reviews/issues/2367) as I go along. This concerns the installation.

The installation process is rather painless, which is great. I have only a few comments:

* [Installation instructions for BlueHive](https://hoomd-tf.readthedocs.io/en/latest/compilation.html#bluehive-installation) are the first item of the [installation guide](https://hoomd-tf.readthedocs.io/en/latest/installation.html). I think it's worth mentioning what BlueHive is and move the the installation instructions at the end (maybe under a new section "Installing on HPC Clusters"?) in order to avoid confusion for external users.
* The following, under [simple compiling](https://hoomd-tf.readthedocs.io/en/latest/compilation.html#simple-compiling), is a bit misleading: "This method assumes you already have installed hoomd-blue and tensorflow. You could do that, for example, via `conda install -c conda-forge hoomd==2.5.2.`". I suggest to remove the last phrase or add also the command to install TensorFlow and `tbb-devel` (see next item).
* I think a clearer note on the need of `tbb-devel` instead of `tbb` would be useful, since `hoomd` installs `tbb` (and not `tbb-devel`) via conda.
* There are many`-Wsign-compare` and some `-Wreorder` warnings when compiling. It would be nice to get rid of them (discussed in #130)

I also wanted to suggest that a `conda` release (as for HOOMD-blue) will possibly attract a larger user base, but I see that this will be addressed in the future (#96).

---

 [JOSS review](https://github.com/openjournals/joss-reviews/issues/2367)

**Comment by RMeli** [comment]
[Unit tests](https://hoomd-tf.readthedocs.io/en/latest/installation.html#unit-tests) have a spurious backslash. I think it should be
```
pytest htf/test-py
```

**Comment by whitead** [comment]
Hi @RMeli! Thank you for doing our review and we appreciate your helpful suggestions! We've added the following PRs to address your notes:

1. #187 revised the documentation based on your notes on `tbb-devel`, the simple compiling, and our HPC install
2.  #189 Addresses all compiler warnings
3. #186 Fixes unit tests, aside from the one you mentioned (MDAnalysis), which we'll address via #190

**Comment by whitead** [comment]
Also, please refer to this [doc website](https://hoomd-tf.readthedocs.io/en/readthedocs/). We'll update the `latest` version once we're done and have bumped the version.

**Comment by RMeli** [comment]
Thanks, LGTM. I'll close this, given #190. 

> Also, please refer to this doc website. We'll update the latest version once we're done and have bumped the version.

Sure, not problem!

**Comment by malramsay64** [comment]
Thanks @RMeli for opening this issue, I was also going to suggest similar changes. @whitead the install instructions are much clearer particularly for the simple case.

**Comment by RMeli** [comment]
@malramsay64 If you have other suggestions feel free to post them here and I'll re-open the issue (in order not to clutter the issue tracker with too many issues for the JOSS review).

**Comment by malramsay64** [comment]
@RMeli I have a couple more points for the installation since I have found a range of situations where the installation could use some clarification.

[ ] gcc 7 is not listed as a dependency. With the major version appearing to be required for compilation, to match the version used to compile tensorflow
[ ] There is no list of supported python versions. `pip install tensorflow==1.15` doesn't work for python 3.8.
[ ] In the instructions for running the tests, it should be a noted that `pytest` and `MDAnalysis` need to be installed to run all the tests.

Also not strictly required, but it might help readability to include the code snippet for the installation of tensorflow in the [simple installation step](https://hoomd-tf.readthedocs.io/en/readthedocs/compilation.html#simple-compiling) as you have done for hoomd.

**Comment by whitead** [comment]
Hi @malramsay64, thanks for doing this review. I've updated the [documentation](https://hoomd-tf.readthedocs.io/en/readthedocs/compilation.html) with your feedback. Specifically:

  1. gcc 7 is not required, just the compiler used to compile tensorflow must be available on the machine. I added a note about this.
  2. Same for python versions, added a note on this too. 
  3. `pytest` note added. The `MDAnalysis` was noted by @RMeli and we'll get it done with #190.

Thanks and let me know if you have any other issues. 

I put the code snippet in the simple compiling section as suggested.
